lib/firehose/server/publisher.rb

```diff
@@ -118,7 +118,7 @@ def eval_publish_script(channel_key, message, ttl, buffer_size, persist, deferra
         local channel_key       = KEYS[3]
         local ttl               = KEYS[4]
         local message           = KEYS[5]
-        local buffer_size       = KEYS[6]
+        local buffer_size       = KEYS[6] + 0
```

Comment:
Lua typecasts on arithmetic, but not on comparison. This forces the conversion to a number ahead of time so we don't have to do it twice.